tests/hpc/barrier_init.pm

```diff
@@ -0,0 +1,67 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017-2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Initialization of barriers for HPC multimachine tests
+# Maintainer: Petr Cervinka <pcervinka@suse.com>
+
+use base 'opensusebasetest';
+use strict;
+use testapi;
+use lockapi;
+use utils;
+
+sub run {
+    # Get number of nodes
+    my $nodes = get_required_var("CLUSTER_NODES");
+
+    # Initialize barriers
+    if (check_var("HPC", "slurm")) {
+        barrier_create("SLURM_MASTER_SERVICE_ENABLED", $nodes);
+        barrier_create("SLURM_SLAVE_SERVICE_ENABLED",  $nodes);
+        barrier_create("SLURM_SETUP_DONE",             $nodes);
+        barrier_create('SLURM_MASTER_RUN_TESTS',       $nodes);
+    }
+    elsif (check_var("HPC", "mrsh")) {
+        barrier_create("MRSH_INSTALLATION_FINISHED", $nodes);
+        barrier_create("MRSH_KEY_COPIED",            $nodes);
```

Comment:
Why are you replacing mutex with barrier for MRSH_KEY_COPIED?

Reply:
Old code was mix of barriers and mutex locks and I want to use only barriers. Is it wrong?

Reply:
If you use barriers everywhere it might get messier and you are loosing information about what exactly are you waiting for. But that's personal preference, if you prefer barriers go for it :)

Reply:
Those are interesting points and I will think about it, but for now I would like to stay with my personal preference (maybe time will teach me). Could you please point me on documentation about it? It looks like https://github.com/os-autoinst/openQA/blob/master/docs/WritingTests.asciidoc doesn't say much about it.

Reply:
Documentation about that is common joke here :) - sorry, but I think you found it all..